markupsafe recipe broken build ('distutils.core' is not a package)

<!--
The issue tracker is a tool to address bugs.
Please use the Discord community or Stack Overflow for support questions,
more information at https://github.com/kivy/kivy-ios#support

Before opening a new issue, make sure you do the following:
    * check that your issue isn't already filed: https://github.com/kivy/kivy-ios/issues
    * prepare a short, runnable example that reproduces the issue
    * reproduce the problem with the latest development version (`master`)
    * double-check that the issue is indeed a bug and not a support request
    * please use backticks to format code or logs
-->

**Versions**

* Python : 3.7
* MacOS version : Catalina
* XCode Version : 11.4.1
* Cython version : 0.28.1

**Describe the bug**
Building the `markupsafe` recipe on the ci via `rebuild_updated_recipes.py` fails.

**To Reproduce**
```
python toolchain.py build python3 markupsafe
```

**Expected behavior**
Green build

**Logs**
```
...
2020-05-02T23:02:51.7088390Z [INFO    ] Running Shell: /Users/runner/runners/2.169.1/work/kivy-ios/kivy-ios/dist/hostpython3/bin/python ('setup.py', 'install', '--prefix', '/Users/runner/runners/2.169.1/work/kivy-ios/kivy-ios/dist/root/python') {'_env': {'CC': '/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/tmp3hkj50t9', 'CXX': '/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/tmpv5l82qfm', 'AR': '/Applications/Xcode_11.4.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ar', 'LD': '/Applications/Xcode_11.4.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld', 'OTHER_CFLAGS': '-I/Users/runner/runners/2.169.1/work/kivy-ios/kivy-ios/dist/include/x86_64/freetype -I/Users/runner/runners/2.169.1/work/kivy-ios/kivy-ios/dist/include/x86_64/hostlibffi -I/Users/runner/runners/2.169.1/work/kivy-ios/kivy-ios/dist/include/x86_64/ffi -I/Users/runner/runners/2.169.1/work/kivy-ios/kivy-ios/dist/include/x86_64/openssl -I/Users/runner/runners/2.169.1/work/kivy-ios/kivy-ios/dist/include/common/sdl2 -I/Users/runner/runners/2.169.1/work/kivy-ios/kivy-ios/dist/include/common/sdl2_image -I/Users/runner/runners/2.169.1/work/kivy-ios/kivy-ios/dist/include/common/sdl2_mixer -I/Users/runner/runners/2.169.1/work/kivy-ios/kivy-ios/dist/include/common/sdl2_ttf -I/Users/runner/runners/2.169.1/work/kivy-ios/kivy-ios/dist/include/x86_64', 'OTHER_LDFLAGS': '-L/Users/runner/runners/2.169.1/work/kivy-ios/kivy-ios/dist/lib', 'CFLAGS': '-O3 -miphoneos-version-min=8.0 -I/Users/runner/runners/2.169.1/work/kivy-ios/kivy-ios/dist/include/x86_64/freetype -I/Users/runner/runners/2.169.1/work/kivy-ios/kivy-ios/dist/include/x86_64/hostlibffi -I/Users/runner/runners/2.169.1/work/kivy-ios/kivy-ios/dist/include/x86_64/ffi -I/Users/runner/runners/2.169.1/work/kivy-ios/kivy-ios/dist/include/x86_64/openssl -I/Users/runner/runners/2.169.1/work/kivy-ios/kivy-ios/dist/include/common/sdl2 -I/Users/runner/runners/2.169.1/work/kivy-ios/kivy-ios/dist/include/common/sdl2_image -I/Users/runner/runners/2.169.1/work/kivy-ios/kivy-ios/dist/include/common/sdl2_mixer -I/Users/runner/runners/2.169.1/work/kivy-ios/kivy-ios/dist/include/common/sdl2_ttf -I/Users/runner/runners/2.169.1/work/kivy-ios/kivy-ios/dist/include/x86_64', 'LDFLAGS': '-arch x86_64 -L/Users/runner/runners/2.169.1/work/kivy-ios/kivy-ios/dist/lib -L/Applications/Xcode_11.4.1.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.4.sdk/usr/lib -miphoneos-version-min=8.0', 'PYTHONPATH': '/Users/runner/runners/2.169.1/work/kivy-ios/kivy-ios/dist/root/python/lib/python2.7/site-packages'}, '_iter': True, '_out_bufsize': 1, '_err_to_out': True}
2020-05-02T23:02:51.7939800Z [DEBUG   ] Traceback (most recent call last):
2020-05-02T23:02:51.7940270Z [DEBUG   ]   File "setup.py", line 9, in <module>
2020-05-02T23:02:51.7940760Z [DEBUG   ]     from distutils.core.command.build_ext import build_ext
2020-05-02T23:02:51.7942270Z [DEBUG   ] ModuleNotFoundError: No module named 'distutils.core.command'; 'distutils.core' is not a package
2020-05-02T23:02:51.8610230Z [INFO    ] cd /Users/runner/runners/2.169.1/work/kivy-ios/kivy-ios/build/markupsafe/arm64/markupsafe-master
2020-05-02T23:02:51.8632800Z Traceback (most recent call last):
2020-05-02T23:02:51.8635670Z   File "toolchain.py", line 1566, in <module>
2020-05-02T23:02:51.8636020Z     ToolchainCL()
2020-05-02T23:02:51.8636370Z   File "toolchain.py", line 1288, in __init__
2020-05-02T23:02:51.8636590Z     getattr(self, args.command)()
2020-05-02T23:02:51.8636950Z   File "toolchain.py", line 1329, in build
2020-05-02T23:02:51.8637170Z     build_recipes(args.recipe, ctx)
2020-05-02T23:02:51.8637430Z   File "toolchain.py", line 1166, in build_recipes
2020-05-02T23:02:51.8637630Z     recipe.execute()
2020-05-02T23:02:51.8638730Z   File "/Users/runner/runners/2.169.1/work/kivy-ios/kivy-ios/toolchain.py", line 749, in execute
2020-05-02T23:02:51.8638940Z     self.build_all()
2020-05-02T23:02:51.8639910Z   File "/Users/runner/runners/2.169.1/work/kivy-ios/kivy-ios/toolchain.py", line 98, in _cache_execution
2020-05-02T23:02:51.8640140Z     f(self, *args, **kwargs)
2020-05-02T23:02:51.8641070Z   File "/Users/runner/runners/2.169.1/work/kivy-ios/kivy-ios/toolchain.py", line 869, in build_all
2020-05-02T23:02:51.8641300Z     self.install()
2020-05-02T23:02:51.8642240Z   File "/Users/runner/runners/2.169.1/work/kivy-ios/kivy-ios/recipes/markupsafe/__init__.py", line 25, in install
2020-05-02T23:02:51.8643080Z     shprint(hostpython, "setup.py", "install", "--prefix", dest_dir, _env=build_env)
2020-05-02T23:02:51.8644460Z   File "/Users/runner/runners/2.169.1/work/kivy-ios/kivy-ios/toolchain.py", line 80, in shprint
2020-05-02T23:02:51.8644810Z     for line in cmd:
2020-05-02T23:02:51.8644950Z   File "tools/external/sh.py", line 565, in next
2020-05-02T23:02:51.8645060Z     self.wait()
2020-05-02T23:02:51.8645170Z   File "tools/external/sh.py", line 500, in wait
2020-05-02T23:02:51.8645270Z     self.handle_command_exit_code(exit_code)
2020-05-02T23:02:51.8645410Z   File "tools/external/sh.py", line 516, in handle_command_exit_code
2020-05-02T23:02:51.8645570Z     raise exc(self.ran, self.process.stdout, self.process.stderr)
2020-05-02T23:02:51.8645660Z sh.ErrorReturnCode_1:
2020-05-02T23:02:51.8645710Z
2020-05-02T23:02:51.8646350Z   RAN: '/Users/runner/runners/2.169.1/work/kivy-ios/kivy-ios/dist/hostpython3/bin/python setup.py install --prefix /Users/runner/runners/2.169.1/work/kivy-ios/kivy-ios/dist/root/python'
2020-05-02T23:02:51.8646430Z
2020-05-02T23:02:51.8646530Z   STDOUT:
2020-05-02T23:02:51.8646620Z Traceback (most recent call last):
2020-05-02T23:02:51.8646770Z   File "setup.py", line 9, in <module>
2020-05-02T23:02:51.8647240Z     from distutils.core.command.build_ext import build_ext
2020-05-02T23:02:51.8647970Z ModuleNotFoundError: No module named 'distutils.core.command'; 'distutils.core' is not a package
2020-05-02T23:02:51.8648100Z
2020-05-02T23:02:51.8648180Z
2020-05-02T23:02:51.8648330Z   STDERR:
2020-05-02T23:02:51.8648410Z
2020-05-02T23:02:51.8835040Z Traceback (most recent call last):
2020-05-02T23:02:51.8835630Z   File ".ci/rebuild_updated_recipes.py", line 27, in <module>
2020-05-02T23:02:51.8835880Z     subprocess.run(f"python3 toolchain.py build {updated_recipes}", shell=True, check=True)
2020-05-02T23:02:51.8836150Z   File "/Users/runner/hostedtoolcache/Python/3.7.6/x64/lib/python3.7/subprocess.py", line 512, in run
2020-05-02T23:02:51.8836350Z     output=stdout, stderr=stderr)
2020-05-02T23:02:51.8837630Z subprocess.CalledProcessError: Command 'python3 toolchain.py build werkzeug pycrypto host_setuptools3 markupsafe hostpython3 python3 itsdangerous kivy jinja2' returned non-zero exit status 1.
2020-05-02T23:02:51.8911590Z - Download 32768 bytes
2020-05-02T23:02:51.8917580Z ##[error]Process completed with exit code 1.
2020-05-02T23:02:51.9047490Z Post job cleanup.
2020-05-02T23:02:52.0745090Z [command]/usr/local/bin/git version
2020-05-02T23:02:52.0942550Z git version 2.26.2
2020-05-02T23:02:52.0999190Z [command]/usr/local/bin/git config --local --name-only --get-regexp core\.sshCommand
2020-05-02T23:02:52.1085390Z [command]/usr/local/bin/git submodule foreach --recursive git config --local --name-only --get-regexp 'core\.sshCommand' && git config --local --unset-all 'core.sshCommand' || :
2020-05-02T23:02:52.2170980Z [command]/usr/local/bin/git config --local --name-only --get-regexp http\.https\:\/\/github\.com\/\.extraheader
2020-05-02T23:02:52.2225130Z http.https://github.com/.extraheader
2020-05-02T23:02:52.2238120Z [command]/usr/local/bin/git config --local --unset-all http.https://github.com/.extraheader
2020-05-02T23:02:52.2310580Z [command]/usr/local/bin/git submodule foreach --recursive git config --local --name-only --get-regexp 'http\.https\:\/\/github\.com\/\.extraheader' && git config --local --unset-all 'http.https://github.com/.extraheader' || :
2020-05-02T23:02:52.3308090Z Cleaning up orphan processes
```

--- a/.ci/constants.py
+++ b/.ci/constants.py
@@ -5,9 +5,6 @@ BROKEN_RECIPES = set(
         "distribute",
         # 'distutils.core' is not a package
         # https://github.com/kivy/kivy-ios/issues/467
-        "markupsafe",
-        # depends on markupsafe
-        # https://github.com/kivy/kivy-ios/issues/466
         "jinja2",
         # bad install directory or PYTHONPATH
         # https://github.com/kivy/kivy-ios/issues/468

--- a/kivy_ios/recipes/markupsafe/__init__.py
+++ b/kivy_ios/recipes/markupsafe/__init__.py
@@ -24,4 +24,5 @@ class MarkupSafeRecipe(PythonRecipe):
         shprint(cmd, "-i", "", "s/features\['speedups'\].*=.*speedups/pass/g", "./setup.py", _env=build_env)  # noqa: W605
         shprint(hostpython, "setup.py", "install", "--prefix", dest_dir, _env=build_env)
 
+
 recipe = MarkupSafeRecipe()

--- a/kivy_ios/recipes/markupsafe/__init__.py
+++ b/kivy_ios/recipes/markupsafe/__init__.py
@@ -16,7 +16,7 @@ class MarkupSafeRecipe(PythonRecipe):
         os.chdir(build_dir)
         hostpython = sh.Command(self.ctx.hostpython)
         build_env = arch.get_env()
-        dest_dir = join(self.ctx.dist_dir, "root", "python")
+        dest_dir = join(self.ctx.dist_dir, "root", "python3")
         build_env['PYTHONPATH'] = join(dest_dir, 'lib', 'python3.8', 'site-packages')
         cmd = sh.Command("sed")
         shprint(cmd, "-i", "", "s/,.*Feature//g", "./setup.py", _env=build_env)

--- a/kivy_ios/recipes/markupsafe/__init__.py
+++ b/kivy_ios/recipes/markupsafe/__init__.py
@@ -20,8 +20,6 @@ class MarkupSafeRecipe(PythonRecipe):
         build_env['PYTHONPATH'] = join(dest_dir, 'lib', 'python3.8', 'site-packages')
         cmd = sh.Command("sed")
         shprint(cmd, "-i", "", "s/,.*Feature//g", "./setup.py", _env=build_env)
-        shprint(cmd, "-i", "", "s/setuptools import Extension/distutils.extension import Extension/g", "./setup.py", _env=build_env)
-        shprint(cmd, "-i", "", "s/setuptools import setup/distutils.core import setup/g", "./setup.py", _env=build_env)
         shprint(cmd, "-i", "", "/^speedups = Feature/,/^)$/s/.*//g", "./setup.py", _env=build_env)
         shprint(cmd, "-i", "", "s/features\['speedups'\].*=.*speedups/pass/g", "./setup.py", _env=build_env)  # noqa: W605
         shprint(hostpython, "setup.py", "install", "--prefix", dest_dir, _env=build_env)

--- a/kivy_ios/recipes/markupsafe/__init__.py
+++ b/kivy_ios/recipes/markupsafe/__init__.py
@@ -26,5 +26,4 @@ class MarkupSafeRecipe(PythonRecipe):
         shprint(cmd, "-i", "", "s/features\['speedups'\].*=.*speedups/pass/g", "./setup.py", _env=build_env)  # noqa: W605
         shprint(hostpython, "setup.py", "install", "--prefix", dest_dir, _env=build_env)
 
-
 recipe = MarkupSafeRecipe()

--- a/kivy_ios/recipes/markupsafe/__init__.py
+++ b/kivy_ios/recipes/markupsafe/__init__.py
@@ -20,7 +20,8 @@ class MarkupSafeRecipe(PythonRecipe):
         build_env['PYTHONPATH'] = join(dest_dir, 'lib', 'python3.8', 'site-packages')
         cmd = sh.Command("sed")
         shprint(cmd, "-i", "", "s/,.*Feature//g", "./setup.py", _env=build_env)
-        shprint(cmd, "-i", "", "s/setuptools/distutils.core/g", "./setup.py", _env=build_env)
+        shprint(cmd, "-i", "", "s/setuptools import Extension/distutils.extension import Extension/g", "./setup.py", _env=build_env)
+        shprint(cmd, "-i", "", "s/setuptools import setup/distutils.core import setup/g", "./setup.py", _env=build_env)
         shprint(cmd, "-i", "", "/^speedups = Feature/,/^)$/s/.*//g", "./setup.py", _env=build_env)
         shprint(cmd, "-i", "", "s/features\['speedups'\].*=.*speedups/pass/g", "./setup.py", _env=build_env)  # noqa: W605
         shprint(hostpython, "setup.py", "install", "--prefix", dest_dir, _env=build_env)

